main.tf

```diff
@@ -85,22 +84,29 @@ data "aws_route53_zone" "default" {
 ## A Route 53 record contains authoritative DNS information for a specified DNS name. DNS records are most commonly used to map a name to an IP Address..
 ##----------------------------------------------------------------------------------
 resource "aws_route53_record" "default" {
-  count = var.enable && var.enable_dns_validation ? 1 : 0
+  for_each = {
+    for record in aws_acm_certificate.cert[0].domain_validation_options[*] : record.domain_name => {
+      name   = record.resource_record_name
+      record = record.resource_record_value
+      type   = record.resource_record_type
+    }
+  }
 
-  zone_id         = join("", data.aws_route53_zone.default[*].zone_id)
-  ttl             = var.ttl
   allow_overwrite = var.allow_overwrite
-  name            = join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_name)
-  type            = join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_type)
-  records         = [join("", aws_acm_certificate.cert[*].domain_validation_options[*].resource_record_value)]
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = var.ttl
+  type            = each.value.type
+  zone_id         = join(",", data.aws_route53_zone.default[*].zone_id)
```

Comment:
Why we are using join function here? Also `[*]` can be replaced with `[0]` 'coz count for hosted zone data is set to 1.

Reply:
because list of string required
![image](https://github.com/clouddrove/terraform-aws-acm/assets/152850912/5c94d496-8b23-41c8-9916-5c7723e8aa5e)

Reply:
if we directly use [*], i.e without join than also the result is list of string, so we can skip using join function.

Reply:
yes, you are right we can skip the join function.